### PR TITLE
Update in `tbl_ard_summary()` to assign default summary types for non-standard ARDs

### LINKED ISSUE #1999: Better `"default_types"` assessment in `tbl_ard_summary()` for diverse data inputs
In the example below, we get an error because the when defining `"default_types"` we don't get a result for non-standard data types.

If we have a non-standard data type, we should inspect the `variable_levels` column of the ARD. If it does not exist or does not have values for the variable, then default to `continuous`. Otherwise default to `categorical` 

``` r
library(survival)
library(cards)
library(gtsummary)
packageVersion("cards")
#> [1] '0.2.2.9009'

# build a table for the survival component
tbl_survival <- 
  # build an ARD for survival
  survfit(Surv(ttdeath, death) ~ trt, trial) |> 
  cardx::ard_survival_survfit(times = c(12, 24)) |> 
  # update the formatting function (requires dev version of cards, or you could do this via `mutate()`)
  cards::update_ard_fmt_fn(
    stat_names = c("estimate", "conf.low", "conf.high"),
    fmt_fn = label_style_sigfig(digits = 2, scale = 100)
  ) |> 
  # dplyr::mutate(context = "categorical") |> # forcing the summary to a "category-like" format
  tbl_ard_summary(
    by = trt,
    label = list(time = "Survival Probability"),
    statistic = time ~ "{estimate}% (95% CI {conf.low}%, {conf.high}%)"
  )
#> Error in if (default_types[[variable]] %in% "continuous" && !type[[variable]] %in% : missing value where TRUE/FALSE needed
```

<sup>Created on 2024-09-25 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* Update in `tbl_ard_summary()` to better handle non-standard ARDs (i.e. not our typical continuous or categorical summaries) by assigning them a default summary type. (#1991)

The most common way to determine how a variable will be summarized (e.g. continuous or categorial) is with the context column. This works well for the standard univariate summaries, but when we incorporate non-standard ARDs, no default type was assigned.

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #1999

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] PR branch has pulled the most recent updates from main branch.
- [x] If a bug was fixed, a unit test was added.
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [x] **All** GitHub Action workflows pass with a :white_check_mark:

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/tbl_ard_summary.R [hunk 1]
```diff
@@ -165,6 +165,15 @@ tbl_ard_summary <- function(cards,
     ) |>
     deframe() |>
     as.list()
+  # for non-standard ARDs, fill in the missing default types
+  for (v in setdiff(include, names(default_types))) {
+    if (!"variable_level" %in% names(cards) ||
+        is_empty(compact(dplyr::filter(cards, .data$variable %in%.env$v)$variable_level))) {
```
Comment (Melkiades) [inline R/tbl_ard_summary.R:171]:
```suggestion
        is_empty(compact(dplyr::filter(cards, .data$variable %in% .env$v)$variable_level))) {
```

File: tests/testthat/test-tbl_ard_summary.R [hunk 2]
```diff
@@ -297,3 +297,40 @@ test_that("tbl_ard_summary() existing 'gts_column'", {
...
+  # ) |>
+  #   dplyr::select(-cards::all_missing_columns()) |>
+  #   dput()
+
+  ard <-
+    structure(list(group1 = c("trt", "trt", "trt", "trt", "trt",
+                              "trt"), group1_level = list("Drug A", "Drug A", "Drug B", "Drug B",
+                                                          "Drug A", "Drug B"), variable = c("time", "time", "time",
+                                                                                            "time", "age", "age"), variable_level = list(12, 24, 12, 24,
+                                                                                                                                         NULL, NULL), context = c("survival", "survival", "survival",
```
Comment (Melkiades) [inline tests/testthat/test-tbl_ard_summary.R:321]: something is off with the styler here

**Review by Melkiades — APPROVED** [review]
All good for me! Thanks Daniel :)

File: tests/testthat/test-tbl_ard_summary.R (hunk 2 above)
Comment (ddsjoberg) [reply tests/testthat/test-tbl_ard_summary.R]: You don't like this style?? 😝😝😝

File: tests/testthat/test-tbl_ard_summary.R [hunk 3]
```diff
@@ -297,3 +297,40 @@ test_that("tbl_ard_summary() existing 'gts_column'", {
...
+  #     dplyr::filter(stat_name %in% c("estimate")) |>
+  #     dplyr::mutate(
+  #       fmt_fn = list("xx.x%"),
+  #       group1_level = unlist(group1_level) |> as.character() |> as.list()
+  #     ),
+  #   cardx::ard_stats_t_test_onesample(trial, variables = age, by = trt) |>
+  #     dplyr::filter(stat_name %in% c("estimate"))
+  # ) |>
+  #   dplyr::select(-cards::all_missing_columns()) |>
+  #   dput()
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply tests/testthat/test-tbl_ard_summary.R]: OH boy, IDK what I was thinking! Updated to just build the ARDs. THANK YOU for questioning haha

**Comment by ddsjoberg** [comment]
Thank you so much @Melkiades !!

**Pull request merged**